api/http/management.go

```diff
@@ -91,27 +74,67 @@ func (h *ManagementHandler) SetSettings(c *gin.Context) {
 		return
 	}
 
-	settings := model.Settings{}
-	if err := c.ShouldBindJSON(&settings); err != nil {
+	integration := model.Integration{}
+	if err := c.ShouldBindJSON(&integration); err != nil {
 		rest.RenderError(c,
 			http.StatusBadRequest,
 			errors.Wrap(err, "malformed request body"),
 		)
 		return
 	}
 
-	// TODO verify that connectionstring has correct permissions
+	// TODO verify that Azure connectionstring / AWS equivalent has correct permissions
 	//      - service
 	//      - registry read/write
 
-	err := h.app.SetSettings(ctx, settings)
+	err := h.app.CreateIntegration(ctx, integration)
 	if err != nil {
-		_ = c.Error(err)
-		rest.RenderError(c,
-			http.StatusInternalServerError,
-			errors.New(http.StatusText(http.StatusInternalServerError)),
-		)
+		switch cause := errors.Cause(err); cause {
+		case app.ErrIntegrationExists:
+			// NOTE: temporary limitation
+			rest.RenderError(c, http.StatusConflict, cause)
+		default:
+			_ = c.Error(err)
+			rest.RenderError(c,
+				http.StatusInternalServerError,
+				errors.New(http.StatusText(http.StatusInternalServerError)),
+			)
+		}
 		return
 	}
 	c.Status(http.StatusNoContent)
 }
+
+// GET /integrations/{id}
+func (h *ManagementHandler) GetIntegration(c *gin.Context) {
+	var (
+		ctx = c.Request.Context()
+		id  = identity.FromContext(ctx)
+	)
+
+	if id == nil || !id.IsUser {
+		rest.RenderError(c, http.StatusForbidden, ErrMissingUserAuthentication)
+		return
+	}
+	integrationID, err := uuid.Parse(c.Param("id"))
+	if err != nil {
+		rest.RenderError(c, http.StatusNotFound, ErrIntegrationNotFound)
```

Comment:
NotFound or BadRequest?

Reply:
I think `NotFound` is more appropriate since it's a path parameter and therefore refers to a resource that cannot exist.